### PR TITLE
[Improve][Zeta] Fix engine e2e error

### PR DESCRIPTION
**Pull request by EricJoy2048**
1. Make all Asynchronous operation use specified thread pool, and not allow to use the thread pool created by jdk (`ForkJoinPool.commonPool()`)
2. Update deleteCheckpoint and support batch delete checkpoint.
3. Enable zeta e2e test.
4. Reduce the queue size(100000 update to 2048) used between source task and sink task, because it may cause oom.
5. Fix the zeta scheduler bug.
6. Fix the zeta checkpoint bug.

### CODE REVIEW

File: config/jvm_options [hunk 1]
```diff
@@ -102,4 +102,7 @@
 8:-Xloggc:${loggc}
 8:-XX:+UseGCLogFileRotation
 8:-XX:NumberOfGCLogFiles=32
-8:-XX:GCLogFileSize=64m
\ No newline at end of file
+8:-XX:GCLogFileSize=64m
+
+-XX:+HeapDumpOnOutOfMemoryError
+-XX:HeapDumpPath=/data/whaletunnel/
```
Comment (liugddx) [inline config/jvm_options]:
```suggestion
-XX:HeapDumpPath=/data/seatunnel/
```

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ public SourceReader<T, SourceSplitBase> createReader(SourceReader.Context reader
         // create source config for the given subtask (e.g. unique server id)
         C sourceConfig = configFactory.create(readerContext.getIndexOfSubtask());
         BlockingQueue<RecordsWithSplitIds<SourceRecords>> elementsQueue =
-                new LinkedBlockingQueue<>(1024);
+                new LinkedBlockingQueue<>(2);
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java:137]: Why is 2

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/reader/SourceReaderBase.java [hunk 3]
```diff
@@ -158,7 +158,12 @@ private RecordsWithSplitIds<E> getNextFetch(Collector<T> output) {
         splitFetcherManager.checkErrors();
         RecordsWithSplitIds<E> recordsWithSplitId = elementsQueue.poll();
         if (recordsWithSplitId == null || !moveToNextSplit(recordsWithSplitId, output)) {
-            log.trace("Current fetch is finished.");
+            try {
+                log.trace("Current fetch is finished.");
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/reader/SourceReaderBase.java]: Unified exception?

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java [hunk 4]
```diff
@@ -339,7 +340,12 @@ private static PreparedStatement initStatement(JdbcConnection jdbc, String sql,
             throws SQLException {
         final Connection connection = jdbc.connection();
         connection.setAutoCommit(false);
-        final PreparedStatement statement = connection.prepareStatement(sql);
+        final PreparedStatement statement =
+                connection.prepareStatement(
+                        sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+        if (fetchSize == 0) {
+            fetchSize = 1024;
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java]:
```suggestion
            fetchSize = Integer.MIN_VALUE;
```

Comment (liugddx) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java]: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-implementation-notes.html

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 5]
```diff
@@ -810,56 +830,77 @@ public void testStreamJobRestoreInAllNodeDown()
...
+            log.info(
+                    "==========================================All node is done========================================");
             Thread.sleep(10000);
 
             node1 = SeaTunnelServerStarter.createHazelcastInstance(seaTunnelConfig);
 
             node2 = SeaTunnelServerStarter.createHazelcastInstance(seaTunnelConfig);
 
+            System.out.println(
+                    "==========================================All node is start, begin check node size ========================================");
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
            log.info(
                    "==========================================All node is start, begin check node size ========================================");
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java [hunk 6]
```diff
@@ -242,8 +243,8 @@ public PipelineState getCheckpoint(String jobId, String pipelineId, String check
         String path = getStorageParentDirectory() + jobId;
         List<String> fileNames = getFileNames(path);
         if (fileNames.isEmpty()) {
-            throw new CheckpointStorageException(
-                    "No checkpoint found for job, job id is: " + jobId);
+            log.info("No checkpoint found for job, job id is: " + jobId);
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java]:
```suggestion
            log.info("No checkpoint found for this job,  the job id is: " + jobId);
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java [hunk 7]
```diff
@@ -207,8 +208,8 @@ public List<PipelineState> getCheckpointsByJobIdAndPipelineId(String jobId, Stri
...
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java]:
```suggestion
            log.info("No checkpoint found for this job, the job id is: " + jobId);
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java [hunk 8]
```diff
@@ -185,18 +185,19 @@ public PipelineState getLatestCheckpointByJobIdAndPipelineId(String jobId, Strin
...
+            return null;
         }
 
         String latestFileName =
                 getLatestCheckpointFileNameByJobIdAndPipelineId(fileNames, pipelineId);
         if (latestFileName == null) {
-            throw new CheckpointStorageException(
+            log.info(
                     "No checkpoint found for job, job id is: "
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java]:
```suggestion
                    "No checkpoint found for this job, the job id is: "
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java [hunk 9]
```diff
@@ -174,7 +174,7 @@ public List<PipelineState> getLatestCheckpoint(String jobId) throws CheckpointSt
                 });
 
         if (latestPipelineStates.isEmpty()) {
-            throw new CheckpointStorageException("No checkpoint found for job, job id:{} " + jobId);
+            log.info("No checkpoint found for job, job id:{} " + jobId);
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java]:
```suggestion
            log.info("No checkpoint found for this job,  the job id:{} " + jobId);
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java [hunk 10]
```diff
@@ -159,8 +159,8 @@ public List<PipelineState> getLatestCheckpoint(String jobId) throws CheckpointSt
         String path = getStorageParentDirectory() + jobId;
         List<String> fileNames = getFileNames(path);
         if (fileNames.isEmpty()) {
-            throw new CheckpointStorageException(
-                    "No checkpoint found for job, job id is: " + jobId);
+            log.info("No checkpoint found for job, job id is: " + jobId);
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java]:
```suggestion
            log.info("No checkpoint found for this  job, the job id is: " + jobId);
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java [hunk 11]
```diff
@@ -135,8 +135,8 @@ public List<PipelineState> getAllCheckpoints(String jobId) throws CheckpointStor
...
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/src/main/java/org/apache/seatunnel/engine/checkpoint/storage/hdfs/HdfsStorage.java]:
```suggestion
            log.info("No checkpoint found for this job, the job id is: " + jobId);
```

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java (hunk 2 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java:137]: 1024 is too big and may cause oom.

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/reader/SourceReaderBase.java (hunk 3 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/reader/SourceReaderBase.java]: done

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 12]
```diff
@@ -133,6 +133,7 @@ public void testBatchJobRunOkIn2Node() throws ExecutionException, InterruptedExc
                     FileUtils.getFileLineNumberFromDir(testResources.getLeft());
             Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
             System.out.println(engineClient.getJobMetrics(clientJobProxy.getJobId()));
+            System.out.println("========================clean test resource====================");
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
            log.info("========================clean test resource====================");
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 13]
```diff
@@ -311,6 +312,8 @@ public void testBatchJobRestoreIn2NodeWorkerDown()
                                     Assertions.assertEquals(
                                             2, finalNode.getCluster().getMembers().size()));
 
+            System.out.println(
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
            log.info(
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 14]
```diff
@@ -345,10 +348,12 @@ public void testBatchJobRestoreIn2NodeWorkerDown()
                             });
 
             // shutdown on worker node
+            System.out.println(
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
            log.info(
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 15]
```diff
@@ -559,13 +565,18 @@ public void testBatchJobRestoreIn2NodeMasterDown()
...
+                    .atMost(600000, TimeUnit.MILLISECONDS)
                     .untilAsserted(
-                            () ->
-                                    Assertions.assertTrue(
-                                            objectCompletableFuture.isDone()
-                                                    && JobStatus.FINISHED.equals(
-                                                            objectCompletableFuture.get())));
+                            () -> {
+                                Thread.sleep(2000);
+                                System.out.println(
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
                                log.info(
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 16]
```diff
@@ -810,56 +830,77 @@ public void testStreamJobRestoreInAllNodeDown()
...
+                    CompletableFuture.supplyAsync(newClientJobProxy::waitForJobComplete);
+
             Awaitility.await()
-                    .atMost(360000, TimeUnit.MILLISECONDS)
+                    .atMost(600000, TimeUnit.MILLISECONDS)
                     .untilAsserted(
                             () -> {
                                 // Wait job write all rows in file
                                 Thread.sleep(2000);
                                 System.out.println(
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
                                log.info(
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java [hunk 17]
```diff
@@ -368,18 +367,23 @@ public void testTwoPipelineBatchJobRestoreIn2NodeWorkerDown()
...
                     .untilAsserted(
-                            () ->
-                                    Assertions.assertTrue(
-                                            objectCompletableFuture.isDone()
-                                                    && JobStatus.FINISHED.equals(
-                                                            objectCompletableFuture.get())));
+                            () -> {
+                                // Wait some tasks commit finished
+                                Thread.sleep(2000);
+                                System.out.println(
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java]:
```suggestion
                                log.info(
```

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java [hunk 18]
```diff
@@ -592,13 +604,19 @@ public void testTwoPipelineBatchJobRestoreIn2NodeMasterDown()
...
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java]:
```suggestion
                                log.info(
```

File: config/jvm_options (hunk 1 above)
Comment (EricJoy2048) [reply config/jvm_options]: done

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java]: Thanks , done

**Review by TyrantLucifer — APPROVED** [review]

**Comment by TyrantLucifer** [comment]
Contained in #4256

**Pull request closed without merging**